src/distributions/binomial.rs

```diff
@@ -129,17 +129,32 @@ mod test {
     use distributions::Distribution;
     use super::Binomial;
 
+    fn test_binomial_mean_and_variance(n: u64, p: f64) {
+        let binomial = Binomial::new(n, p);
+        let mut rng = ::test::rng(123);
+
+        let expected_mean = n as f64 * p;
+        let expected_variance = n as f64 * p * (1.0 - p);
+
+        let mut results = [0.0; 1000];
+        for i in results.iter_mut() { *i = binomial.sample(&mut rng) as f64; }
+
+        let mean = results.iter().sum::<f64>() / results.len() as f64;
+        assert!((mean as f64 - expected_mean).abs() < expected_mean / 50.0);
+
+        let variance =
+            results.iter().map(|x| (x - mean) * (x - mean)).sum::<f64>()
+            / (results.len() - 1) as f64;
```

Comment:
I believe the formula for variance is ((x-mean)^2)/(n-1).

Reply:
Where did you get that from? It should be over n; you are simply averaging each sample's variance. See https://en.wikipedia.org/wiki/Variance#Discrete_random_variable

Also, it is arguable that you should be using the expected mean instead of the actual mean here. That might improve the tolerance you can use in the test?

Reply:
Ah, there is a difference between taking the variance of all results, and of only a sample of them. I should know these things better...